src/main/java/uk/gov/hmcts/darts/hearings/mapper/GetEventsResponseMapper.java

```diff
@@ -26,5 +33,30 @@ public EventResponse mapToEvent(EventEntity eventEntity) {
         return eventResponse;
     }
 
+    public List<EventEntity> filterNonLatestEvents(List<EventEntity> events) {
+
+        var groupedEvents =  events.stream()
+            .collect(Collectors.groupingBy(event -> Optional.ofNullable(event.getEventId())));
+
+        return groupedEvents.values().stream()
+            .flatMap(GetEventsResponseMapper::getLatestEvent)
+            .filter(Objects::nonNull)
+            .toList();
+    }
+
+    private Stream<EventEntity> getLatestEvent(List<EventEntity> group) {
+        if (group.get(0).getEventId() == null ||  group.get(0).getEventId() == 0) {
```

Comment:
Does this logic assume the incoming list of events is sorted by timestamp? If so, there's a risk of picking up the wrong event if the calling code doesn't ensure the data is pre-sorted. Consider explictly sorting the list here (or in `filterNonLatestEvents()`) before doing the conditional checks.

Reply:
After discussion with @jackmaloney , the above comment can be ignored. The order at this point does not matter - the EventIds will all be identical since they've been grouped within `filterNonLatestEvents()`.